### ISSUE TITLE
v7 horizontal scroll bar absent in Firefox

### ISSUE THREAD

**Issue by liquidcarbon** [jupyter/notebook#6960]
## Firefox - no way to scroll to the last columns

![image](https://github.com/jupyter/notebook/assets/47034358/bbbd73fe-0f1c-4484-bbe9-269e93bb10e9)

## Edge

![image](https://github.com/jupyter/notebook/assets/47034358/ba728a20-4460-486a-82a6-26bfed6d8405)

**Comment by svaningelgem** [comment]
As mentioned in ticket #14625 of jupyterlab for googlers arriving here:

Workaround to show the scrollbars from within a Notebook cell:
```python
from IPython.display import display, HTML
display(HTML("<style>.jp-OutputArea-output {display:flex}</style>"))
```

**Comment by svaningelgem** [comment]
I can confirm @liquidcarbon 's observations:
- edge shows horizontal bars
- chrome shows them as well
- firefox doesn't

**Comment by krassowski** [comment]
@svaningelgem

When you see this issue with Notebook v7.0.5+:
1. Can you reproduce in a clean environment with a fresh pip install?
2. What version of Firefox you use?
3. Can you shift + right-click over the data frame output and select "inspect" and create a screenshot like below:
![Screenshot from 2023-10-17 18-17-05](https://github.com/jupyter/notebook/assets/5832902/ba562405-02cf-43d7-9624-10583cd380f6)

**Comment by svaningelgem** [comment]
I just installed a new environment with this command:
```bash
mamba create -n v7bar python=3.11 notebook ipywidgets pandas
```
This installed v7.0.6 of notebook.

I then started the notebook:
```bash
conda activate v7bar
jupyter notebook
```
I then proceeded with this cell:
```python
import numpy as np
import pandas as pd
df = pd.DataFrame(np.arange(100))
with pd.option_context('display.max_columns',100):
    display(df.T)
```

And ... I do have a scrollbar.

Ok, so I'm now off investigating what is different between the 2 environments.
FYI my screenshot resembles yours:
![image](https://github.com/jupyter/notebook/assets/199434/fab765ea-a06e-4843-8391-7e749ece300c)

And this is in the other notebook where I don't have the scrollbar:
![image](https://github.com/jupyter/notebook/assets/199434/35f61b48-0fbe-4554-8d22-5fa277c36fb3)

**Comment by krassowski** [comment]
Can you add a screenshot showing Help → About window in the faulty environment? My suspicion would be two different versions of notebook installed (one via conda, one via pip) or in a very unlikely event a problem with conda-forge assets.

**Comment by svaningelgem** [comment]
It gets weirder. Everything worked fine with the new kernel (v7bar).
So I went back to the old one (howest), and tried it on an empty notebook. Worked fine.
I restarted the kernel (from within E:\tmp), opened the demo notebook: fine
I copied the notebook into that directory and opened it: fine

But... I figured out now how to reproduce it. I've no clue how, why or what, but I can reproduce it.

1. `mamba create -n v7bar python=3.11 notebook ipywidgets pandas`
2. open a cmd
3. cd e:\tmp
4. jupyter notebook (port for me: 8889)
5. open a notebook (or create a new one if it isn't there yet)
6. copy/paste the pandas code from above to simulate a large dataframe
7. run it
8. ===> No scrollbars
9. save the document (I use CTRL-S)
10. open another cmd
11. cd e:\tmp
12. jupyter notebook (port for me: 8890)
13. open the previously created notebook
14. ====> scrollbars!

No clue... This is by most the weird behavior I've ever noticed 😁. And I'm already working for 20 years in software development 🤣.

**Comment by svaningelgem** [comment]
> Can you add a screenshot showing Help → About window in the faulty environment? My suspicion would be two different versions of notebook installed (one via conda, one via pip) or in a very unlikely event a problem with conda-forge assets.

So basically, both environments are fine, as long as they're not the first kernel started...

**Comment by svaningelgem** [comment]
I now tried it another way:
```bash
mamba create -n v7bar python=3.11
mamba activate v7bar
pip install jupyter notebook pandas
```

Same behavior: 1st kernel I start has no scrollbars, 2nd kernel has.

So I doubt it has something to do with conda vs pip assets.
For completeness sake:
```
(v7bar) e:\tmp>pip list | grep notebook
notebook                  7.0.6
notebook_shim             0.2.3

(v7bar) e:\tmp>pip list | grep jupyter
jupyter                   1.0.0
jupyter_client            8.4.0
jupyter-console           6.6.3
jupyter_core              5.4.0
jupyter-events            0.8.0
jupyter-lsp               2.2.0
jupyter_server            2.8.0
jupyter_server_terminals  0.4.4
jupyterlab                4.0.7
jupyterlab-pygments       0.2.2
jupyterlab_server         2.25.0
jupyterlab-widgets        3.0.9
```
About window:
![image](https://github.com/jupyter/notebook/assets/199434/f5055f3d-a10c-4a62-997f-610bceb82550)

**Comment by svaningelgem** [comment]
Small additional remark:
I have thus 2 environments: howest & v7bar. When I start the first kernel in the first environment, I don't have a scrollbar. When I start the second kernel in the 2nd environment, I do have one.
So related to the port of the server or such??

Hmmm, changing the default port to 7777 via:
```python
c.NotebookApp.port = 7777
```

Makes the scroll bars appear in the first kernel.
Setting the port to 8890: check. Works fine.
Setting the port to 8889: gone...

W..T..F.. (I think my current unbelief warrants this outburst 😁)

**Comment by krassowski** [comment]
Could it be that there is an old server with old version of notebook running in the background and you need to restart you computer to get rid of it?

**Comment by svaningelgem** [comment]
That suggestion makes sense. So I just restarted my computer. As I have IntelliJ and WSL2 installed, where both could -independently- start up another version of the jupyter server.
```bash
(.venv) ~$ ps afx | grep notebo
    441 pts/0    S+     0:00  |           \_ grep --color=auto notebo
(.venv) ~$ ps afx | grep jupyt
    449 pts/0    S+     0:00  |           \_ grep --color=auto jupyt

> tasklist | grep -i jupyt

> tasklist | grep -i notebook

> net stop iphlpsvc
```
So nothing running, and `iphlpsvc` was running on port 8888.

Then:
- started a jupyter notebook. It opened a browser tab to 8888. >> scrollbars
- I started another kernel. A browser tab opened to port 8889 >> no scrollbars
- I opened up another kernel. A browser tab opened to port 8890 >> scrollbars

So weird!

**Comment by krassowski** [comment]
What do you mean be starting another kernel? Can you provide a full screencast? Given the information about WSL, it could be that you have the old version installed elsewhere.

**Comment by svaningelgem** [comment]
Last, but not least... I'm going to delete my history, cache, cookies and everything. Will update this comment.

**Comment by jtpio** [comment]
@svaningelgem just checking if you are still seeing this issue with the latest versions?

Trying with the latest `7.1.0b0` (still pre-release for now), it seems to be fine with Firefox 122?

[notebook-horizontal-scrollbar.webm](https://github.com/jupyter/notebook/assets/591645/b935be2b-9bba-4962-8587-bf2055b1de17)

**Comment by jtpio** [comment]
Closing as fixed in 7.1.

Please feel free to open a new issue if needed, thanks!
